### PR TITLE
Encouraging Contributors

### LINKED ISSUE #1097: Encouraging Contributors to Code Review
It can sometimes take time for maintainers to review a PR, and if a PR goes through several iterations of maintainer review, the lifetime of the PR is extended even further.  If contributors are willing to review PRs, it can oftentimes accelerate the review process as it allows the PR to be in a more finished state before maintainers review, and fewer maintainer review iterations are needed.  It has always been appreciated when contributors are willing to review each others' PRs, but this has not been explicitly said anywhere.  

We should make this clear on the wiki or in the README so contributors may be more comfortable reviewing open PRs.

### PR DESCRIPTION
**Pull request by AlexGuteniev**
My take on resolve #1097

### CODE REVIEW

File: README.md
Comment (jovibor) [inline README.md:18]:
```suggestion
about what we're working on. You can also submit pull requests to fix bugs or add features. See [CONTRIBUTING.md][] for

```

File: CONTRIBUTING.md
Comment (mnatsuhara) [inline CONTRIBUTING.md:21]: Just adding a few sentences to expand on the rationale here :) 
```suggestion
We welcome code reviews from contributors! Reviews from other contributors can often accelerate the reviewing process by helping a PR reach a more finished state before maintainers review the changes.  As a result, such a PR may require fewer maintainer review iterations before reaching a "Ready to Merge" state!  
```

File: CONTRIBUTING.md
Comment (mnatsuhara) [inline CONTRIBUTING.md:22]:
```suggestion
To gain insight into our Code Review process, you may want to view:
```

File: CONTRIBUTING.md
Comment (mnatsuhara) [inline CONTRIBUTING.md:32]: Do we often link to https://eel.is/c++draft in official documentation for the repo?  I have seen it cited in PR comments and issues on occasion, but since it's technically an unofficial source...
> (Generated on 2021-11-07 from the LaTeX sources by cxxdraft-htmlgen. This is not an ISO publication.
For historical versions of the document, see Tim Song's cppwp page.)
> Note: this is an early draft. It's known to be incomplet and incorrekt, and it has lots of bad formatting.

...I wonder if we should instead cite a specific working draft?

File: CONTRIBUTING.md
Comment (mnatsuhara) [inline CONTRIBUTING.md:45]: Would this be clearer if we split it into a list of acceptable sources vs. unacceptable?  
```suggestion
* Your changes are written from scratch using only acceptable sources: 
   * this repository
   * the C++ Working Draft (including any cited standards)
   * other WG21 papers (excluding
  reference implementations outside of proposed standard wording)
   * LWG issues as reference material
   * a project listed in NOTICE.txt (please cite the project in the PR description)
  If your changes are derived from any other project, you *must* mention it
  here, so we can determine whether the license is compatible and whether any other 
  steps need to be taken
```

File: CONTRIBUTING.md
Comment (AlexGuteniev) [reply CONTRIBUTING.md:8]: I don't agree. Building and consuming doesn't necessarily mean contribution intention.

File: CONTRIBUTING.md
Comment (AlexGuteniev) [reply CONTRIBUTING.md:32]: I think we should go with [[lex.name]/3.1](https://eel.is/c++draft/lex.name#3.1)

Comment (AlexGuteniev) [reply CONTRIBUTING.md:32]: We use stable names in source comments and we do use draft references

File: CONTRIBUTING.md
Comment (AlexGuteniev) [reply CONTRIBUTING.md:8]: Possibly we can move out "reporting an issue" instead.

Comment (mnatsuhara) [reply CONTRIBUTING.md:8]: I saw that you created an issue for this!  Thanks :) We can continue discussion there.

File: CONTRIBUTING.md
Comment (mnatsuhara) [inline CONTRIBUTING.md:27]:
```suggestion
 * [Advice for Reviewing Pull Requests](https://github.com/microsoft/STL/wiki/Advice-for-Reviewing-Pull-Requests)
```

**Review by mnatsuhara — APPROVED** [review]
I think this looks great!  This is something I think we can work on expanding further later on (with the WIP contribution guidelines, documentation about avoiding breaking ABI, and the meta-documentation discussed at the last contributor meetup), but this is good start and is nicely self-contained.  

(The one suggestion I had was removing an inconsistent comma, but that change can wait until the second review so we don't reset testing just for that.)

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:13]: These should probably be `##` and `###` respectively, so as to be placed under the `Contributing` \<h1\>.

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:26]:
```suggestion
 * pull requests which are [undergoing review][review:changes-requested],
```

(with a link block):

```md
[review:changes-requested]: /microsoft/STL/pulls?q=is%3Apr+is%3Aopen+review%3Achanges-requested
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:28]:
```suggestion
 * our [code review videos][].
```

separate link (plus minor grammar)

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:34]:
```suggestion
1. Any non-standard identifiers in product code (including local variables, exposition-only members, etc.)
   begin with an underscore and a capital letter, as per [\[lex.name\]p3.1][]
```

grammar, make the link out-of-line, etc.

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:32]:
```suggestion
Before submitting a pull request, please ensure:
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:36]:
```suggestion
```
no longer necessary.

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:38]:
```suggestion
2. Your changes don't introduce change ABI for any type or function that a user may use (including adding or removing non-static data
```

renaming a member does not break ABI, but removing one does

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:39]:
```suggestion
  members, adding or removing virtual member functions, changing whether a type is an aggregate
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:42]:
```suggestion
3. Your changes are written from scratch using only acceptable sources:
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:47]:
```suggestion
   * LWG issues,
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:48]:
```suggestion
   * a project listed in [NOTICE.txt][] (make sure to cite the project in the PR description!).
```

make sure to add the link block

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:43]:
```suggestion
   * this repository,
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:44]:
```suggestion
   * the C++ Working Draft (including any cited standards),
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:46]:
```suggestion
  reference implementations outside of proposed standard wording),
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:49]:
```suggestion
  If your changes are derived from any other project, you _must_ mention it
```

I like `_` for italicizing

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
Mostly grammar and nits

File: CONTRIBUTING.md
Comment (AlexGuteniev) [reply CONTRIBUTING.md]: Was not able to make it non-inline link

File: CONTRIBUTING.md
Comment (AlexGuteniev) [reply CONTRIBUTING.md]: I'm not really sure if it should be changes requested state. Maybe review requested state?

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:35]:
```suggestion
1. Any non-standard identifiers in product code (including local variables, exposition-only members, etc.).
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
minor formatting

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:51]:
```suggestion
If your changes are derived from any other project, you _must_ mention it
in the pull request description, so we can determine whether the license is compatible and whether any other
steps need to be taken.
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:35]:
```suggestion
1. Any non-standard identifiers in product code (including local variables, exposition-only members, etc.)
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:36]:
```suggestion
   begin with an underscore and a capital letter, as per [\[lex.name\]/3.1][].
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:66]:
```suggestion
[wiki:videos]: https://github.com/microsoft/STL/wiki/Code-Review-Videos
[\[lex.name\]/3.1]: https://eel.is/c++draft/lex.name#3.1
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:47]:
```suggestion
   * a project listed in [NOTICE.txt][] (make sure to cite the project in the PR description!).
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:29]:
```suggestion
 * [Advice for Reviewing Pull Requests][wiki:advice-for-reviewing], and
 * [Code Review Videos][wiki:videos].
```

File: CONTRIBUTING.md
Comment (strega-nil-ms) [inline CONTRIBUTING.md:46]:
```suggestion
   * LWG issues, or
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]

**Review by strega-nil-ms — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Looks good! I pushed a conflict-free merge with `main`, followed by one grammar fix and minor whitespace changes.

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for improving our guidance for contributors! :page_facing_up: :smile_cat: :tada:

**Pull request merged**
